### ISSUE TITLE
Move reseed + download impl out of core and into client

### ISSUE THREAD

**Issue by anonimal** [monero-project/kovri#329]
**By submitting this issue, I confirm the following:**

- I have read and understood the [contributor guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.

*Place an X inside the bracket to confirm*
- [X] I confirm.

---

As I've been working on #305 which touches on #168, #48, and #98, as discussed with @EinMByte in ```#kovri-dev```: [discussion](https://github.com/monero-project/kovri/files/441349/2016-08-28_library_discussion.txt) held after today's meeting.

TL;DR: client implementations should handle all downloading.
